packages/api/src/controllers/clip.ts

```diff
@@ -0,0 +1,154 @@
+import { validatePost } from "../middleware";
+import { Request, Router } from "express";
+import _ from "lodash";
+import { db } from "../store";
+import { NotFoundError } from "../store/errors";
+import { pathJoin } from "../controllers/helpers";
+import {
+  createAsset,
+  validateAssetPayload,
+  defaultObjectStoreId,
+  catalystPipelineStrategy,
+} from "./asset";
+import { generateUniquePlaybackId } from "./generate-keys";
+import { v4 as uuid } from "uuid";
+import { DBSession } from "../store/db";
+import { fetchWithTimeout } from "../util";
+import { DBStream } from "../store/stream-table";
+
+const app = Router();
+
+app.post("/", validatePost("clip-payload"), async (req, res) => {
+  const playbackId = req.body.playbackId;
+  const userId = req.user.id;
+
+  const id = uuid();
+  let uPlaybackId = await generateUniquePlaybackId(id);
+
+  const content = await db.stream.getByPlaybackId(playbackId); //||
+  //(await db.asset.getByPlaybackId(playbackId));
+
+  let isStream: boolean;
+  if (content && "streamKey" in content) {
+    isStream = true;
+  }
+
+  if (!content) {
+    throw new NotFoundError("Content not found");
+  }
+
+  const user = await db.user.get(content.userId);
+
+  if (!user || userId !== content.userId) {
+    throw new NotFoundError("Content not found");
+  }
+
+  if ("suspended" in content && content.suspended) {
+    throw new NotFoundError("Content not found");
+  }
+
+  let asset = await validateAssetPayload(
+    id,
+    uPlaybackId,
+    content.userId,
+    Date.now(),
+    defaultObjectStoreId(req),
+    req.config,
+    {
+      name: req.body.name || `clip_${uPlaybackId}`,
+    },
+    // TODO: actual type `clip` - currently something's wrong with the types when setting up the schema
+    { type: "directUpload", sourceId: content.id }
```

Comment:
This should be **really** easy to implement so please don't skip it on this first PR! Just add the enum on the schema and set it here.

Additionally, we should differentiate between streams/sessions and asset IDs, because our IDs have become non-globally-unique since Recordings V2 (sessions and assets have the same ID).

So I'd suggest sth like this
```suggestion
    { type: "clip", ...(isStream, ? { sessionId: session.id } : { assetId: content.id }) }
```
WDYT? (the syntax doesn't have to be like that, just an example)

Reply:
This makes a lot of sense, changed it!